src/main/java/com/microsoft/azure/relay/ClientWebSocket.java

```diff
@@ -344,6 +341,15 @@ public void onError(Session session, Throwable cause) {
 		RelayLogger.throwingException(cause, this);
 	}
 	
+	private static ByteBuffer deepCopyByteBuffer(ByteBuffer original) {
```

Comment:
>deepCopyByteBuffer [](start = 27, length = 18)

If the original ByteBuffer has consumed some bytes (e.g. position > 0) should the clone result in a ByteBuffer that when asked for the next bytes continues at the position of the original ByteBuffer or represent the original ByteBuffer after rewind was called?

Reply:
If I call ClientWebSocket/HybridConnectionStream send/write method which takes a ByteBuffer with position != 0 will the library rewind my ByteBuffer and send bytes which I basically just told it not to?

---
In reply to: [296396267](https://github.com/Azure/azure-relay-java/pull/48#discussion_r296396267) [](ancestors = 296396267)